app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v4.rb

```diff
@@ -551,10 +551,24 @@ def get_mac_address_of_nic_on_requested_vlan(args)
     end
 
     def find_mac_address_on_network(nics, vnic_profile_id)
-      nic = nics.detect do |n|
-        n.vnic_profile.id == vnic_profile_id
+      if vnic_profile_id == '<Empty>'
+        nic = nics.detect do |n|
+          n.vnic_profile.nil?
+        end
+      elsif vnic_profile_id == '<Template>'
+        nic = nics.first
+      else
+        nic = nics.detect do |n|
+          n.vnic_profile.id == vnic_profile_id
+        end
       end
-      _log.warn "Cannot find NIC with vnic_profile=#{vnic_profile_id}" if nic.nil?
+
+      if nics.empty?
+        _log.warn "Cannot find mac address, there are no NICs"
+      else
+        _log.warn "Cannot find mac address, cannot find NIC with vnic_profile=#{vnic_profile_id}" if nic.nil?
```

Comment:
it sounds like there are two error messages in this line: one for mac address and the other for the profile.
Maybe 'or' between the two sentence parts or can we be more specific based on the detected nic ?

Reply:
done